roman/robot.py

```diff
@@ -44,26 +53,53 @@ def step(self, dx, dy, dz, dyaw, gripper_state=hand.Position.OPENED, max_speed =
         '''
         self.arm.read()
         pose = self.arm.state.tool_pose()
-        pose = arm.Tool.from_xyzrpy(pose.to_xyzrpy() + [dx,dy, dz,0,0, dyaw])
-        self.hand.move(position = gripper_state, blocking = False)
-        self.arm.move(pose, max_speed = max_speed, max_acc=max_acc, blocking = False)
-        end = self.arm.state.time() + dt
-        while self.arm.state.time() < end and not (self.arm.state.is_done() and self.hand.state.is_done()):
-            self.read()
+        pose = pose.to_xyzrpy() + [dx, dy, dz, 0, 0, dyaw]
+        x, y, z, _, _, yaw = pose
+        arm_state, gripper_state = self.step_absolute(x, y, z, yaw, gripper_state=gripper_state, touch=touch, max_speed=max_speed, max_acc=max_acc, blocking=blocking, dt = dt)
+        return arm_state, gripper_state
+
+    
+    def step_absolute(self, x, y, z, yaw, gripper_state=hand.Position.OPENED, touch=False, max_speed=0.5, max_acc=0.5, blocking=True, dt=0.2):
+        '''
+        Moves the arm to the specified absolute position in carthesian coordinates, 
+        assuming the gripper is vertical (aligned with the z-axis), pointing down.
+        This version returns after the amount of time specified by dt.
+        This supports the simplest Gym robotic manipulation environment.
```

Comment:
Is this line outdated doc? Should we remove it?

Reply:
Yes, it was outdated. I've updated it to clarify.